junifer/preprocess/bold_warper.py

```diff
@@ -49,9 +59,25 @@ class BOLDWarper(BasePreprocessor):
     def __init__(self, reference: str) -> None:
         """Initialize the class."""
         self.ref = reference
-        super().__init__(
-            on="BOLD", required_data_types=["BOLD", self.ref, "Warp"]
-        )
+        # Initialize superclass based on reference
+        if self.ref == "T1w":
+            super().__init__(
+                on="BOLD", required_data_types=["BOLD", self.ref, "Warp"]
+            )
+        elif self.ref in [
+            "MNI152Lin",
```

Comment:
Doesn't this depend on the available tranformations? Shall we encapsulate this somewhere?

Reply:
I've restricted it for now to not be abused, will relax it once it matures.